Client/Frontend/Home/RemoteTabsPanel.swift

```diff
@@ -82,29 +82,43 @@ class RemoteTabsPanel: UITableViewController, HomePanel {
             return
         }
 
-        // Otherwise, fetch the tabs cloud
-        self.profile.getClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
-            self.refreshControl?.endRefreshing()
+        // Otherwise, fetch the tabs cloud if its been more than 1 minute since last sync
+        let lastSyncTime = profile.prefs.timestampForKey("lastRemoteTabSyncTime")
+        if (lastSyncTime == nil) || NSDate.now() - lastSyncTime! > OneMinuteInMilliseconds {
+            self.profile.getClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
+                if let success = result.successValue {
+                    self.profile.prefs.setTimestamp(NSDate.now(), forKey: "lastRemoteTabSyncTime")
+                }
+                self.updateUIWithClientAndTabs(result)
+            }
+        } else {
+            self.profile.getCachedClientsAndTabs().uponQueue(dispatch_get_main_queue()) { result in
```

Comment:
Idea: rather than `if (stale) { fetch } else { load }`, let's do

```
if (recent) {
  if (load) {
    return
  }
  fetch
}
```

The scenario I'm thinking of is something like Clear Private Data or disconnect/reconnect. This'll recover more gracefully.